src/portable/st/stm32f4/dcd_stm32f4.c

```diff
@@ -80,20 +80,39 @@ static void bus_reset(void) {
   dev->DOEPMSK |= USB_OTG_DOEPMSK_STUPM | USB_OTG_DOEPMSK_XFRCM;
   dev->DIEPMSK |= USB_OTG_DIEPMSK_TOM | USB_OTG_DIEPMSK_XFRCM;
 
+  // Peripheral FIFO architecture (Rev18 RM 29.11)
+  //
+  // --------------- 312.5 ( 1250 bytes )
+  // | IN FIFO 3  |
+  // --------------- y + x + 16 + GRXFSIZ
+  // | IN FIFO 2  |
+  // --------------- x + 16 + GRXFSIZ
+  // | IN FIFO 1  |
+  // --------------- 16 + GRXFSIZ
```

Comment:
Just a side note I think worth mentioning: `IN FIFO 0` does _not_ have to map to endpoint 0 (which is where you got `16` entries from I assume). It's simply the easiest mapping :).

Reply:
Yeah, right, any IN FIFO can be assigned to any IN endpoint, it doesn't need to have fifo index = epnum IN. Currently, we fixed 16 words for control IN which is enough since control doesn't carry much payload, and 80 for each enpoint IN. Later on we can enhance it with better FIFO allocation scheme. e.g Interrupt endpoint only has up to 16 words since Interrupt Endpoint does transfer much, Bulk Endpoint can have as much as possible better is  128 word (512 since it is a sweet spot for example msc class). But we can always improve this later :)